### ISSUE TITLE
[FEATURE] Season "offset"

### ISSUE THREAD

**Issue by CnC-ode** [umbrellaplug/umbrellaplug.github.io#373]
### Feature

Hello,

I have an issue with a show I casually watch. The show is called "Whose Line Is It Anyway? (US)" and it has 2 TMDB ids:

- 61018 (1998–2007) [Whose Line Is It Anyway?](https://www.themoviedb.org/tv/61018-whose-line-is-it-anyway)
- 64978 (2013–2024) [Whose Line Is It Anyway?](https://www.themoviedb.org/tv/64978-whose-line-is-it-anyway)

I tried to use the "Require Year in source for known Reboot shows" option but that didn't work.

As I was digging down the problem, I found out that, because the original show (61018) had 8 seasons, the sources for the reboot (64978) have a season "offset" of 8. For example, reboot season 1 will have sources with "S09EXX" instead of "S01EXX"...

I managed to "solve" the problem with some strategically placed hardcoded lines on sources.py xD

- In order to scrape the right sources, added this on play()
```
season_for_source = str(int(season) + 8) if tmdb == '64978' else season
if rescrape: self.clr_item_providers(title, year, imdb, tmdb, tvdb, season_for_source, episode, tvshowtitle, premiered)
items = providerscache.get(self.getSources, self.providercache_hours, title, year, imdb, tmdb, tvdb, season_for_source, episode, tvshowtitle, premiered)
```

- In order to resolve the correct file to play, added this to sourcesResolve()
```
if meta.get('tmdb') == 64978: season = str(int(season) + 8)
```

Even though this is a niche issue, I though I'd share it because I believe other shows might suffer from the same problem and this is not an elegant solution at all. Pretty sure I will need to add the lines after each addon update.

Maybe you could consider implementing a feature to manage this "offset" based on TMDB id.

Cheers!

**Comment by umbrellaplug** [comment]
Yeah this is where is gets tough. There's exceptions for different shows / movies written into different parts of umbrella. This is likely going to be another one of those situations. There's ton's of issues with different shows and their metadata not matching. Dr Who is a great example. I can't really write every possible scenario for every show but I will look into what I can do for this. It's just always jacked up metadata.

**Comment by CnC-ode** [comment]
Thanks for the quick reply.

Of course, I understand that there must be a ton of different issues like this and I truly congratulate you for taking the time and try to tackle them one by one! I actually did stumble upon some hardcoded stuff like this for other shows in your code. I just thought to share this one as other people may be having the same problem with this show or another that falls into the same scenario.

While I was digging up the problem I had this idea where instead of tackling specific scenarios, a feature could be developed in order to allow custom rules for the end user to manage themselves. For example, in this case it would be "as simple as" a pair of tmdbid & "season offset". Of course there would be much complicated scenarios than this which could not be easily handled with custom rules, but at least you wouldn't need to hardcode stuff every time someone stumbled upon another one.

That said, I managed to find this solution for this specific scenario and I will happily live with it xD

Nevertheless, I'm just giving my 2 cents in the spirit of sharing and contributing for a addon I truly love.

Kudos!

**Comment by umbrellaplug** [comment]
Decided not to implement this one. it's a little too niche.
